### PR TITLE
WT-7569 Fix wrongly squash an out of order timestamp update

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

**Comment by quchenhao** [comment]
mongodb patch build:
https://spruce.mongodb.com/version/60a7182ed1fe076ad7cf65ba/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

**Comment by hbokhari** [comment]
@quchenhao, is it ready for a review?

**Comment by quchenhao** [comment]
@hbokhari @bvpvamsikrishna It is ready to review now.

**Review by agorrod — COMMENTED** [review]
Thanks for the reproducer - I don't believe this is the right fix. The conversation in the JIRA ticket explains.

File: src/history/hs_rec.c [hunk 1]
```diff
@@ -495,11 +495,9 @@ __wt_hs_insert_updates(
...
          */
         modify_cnt = 0;
-        for (; updates.size > 0 &&
-             !(upd->txnid == list->onpage_upd->txnid &&
-               upd->start_ts == list->onpage_upd->start_ts);
-             tmp = full_value, full_value = prev_full_value, prev_full_value = tmp,
-             upd = prev_upd) {
+        for (; updates.size > 0; tmp = full_value, full_value = prev_full_value,
+                                 prev_full_value = tmp, upd = prev_upd) {
+            WT_ASSERT(session, upd != list->onpage_upd);
```
Comment (hbokhari) [inline src/history/hs_rec.c:500]: Can you explain this assert in a comment?

Comment (quchenhao) [reply src/history/hs_rec.c:501]: Added a comment

**Review by hbokhari — APPROVED** [review]
LGTM

Make sure you have run a MongoDB patch build with this one.

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

**Pull request merged**